annotation-server/src/common/google-sheets.spec.ts

```diff
@@ -0,0 +1,121 @@
+        it('should not return an error with valid sheet id', async () => {
+            const spreadsheetData = await fetchSpreadsheetCells(
+                configService.get<string>('GOOGLESHEET_ID'),
+                configService.get<string>('GOOGLESHEET_APIKEY'),
+                [
+                    configService.get<string>('GOOGLESHEET_RANGE_MEDICATIONS'),
+                    configService.get<string>('GOOGLESHEET_RANGE_GENES'),
+                    configService.get<string>('GOOGLESHEET_RANGE_PHENOTYPES'),
+                    configService.get<string>('GOOGLESHEET_RANGE_IMPLICATIONS'),
+                    configService.get<string>(
+                        'GOOGLESHEET_RANGE_RECOMMENDATIONS',
+                    ),
+                ],
+            );
+            expect(spreadsheetData).toBeDefined();
+        });
+
+        it('should return an error with invalid sheet id', async () => {
+            expect(() =>
+                fetchSpreadsheetCells(
+                    '1',
+                    configService.get<string>('GOOGLESHEET_APIKEY'),
+                    [
+                        configService.get<string>(
+                            'GOOGLESHEET_RANGE_MEDICATIONS',
+                        ),
+                        configService.get<string>('GOOGLESHEET_RANGE_GENES'),
+                        configService.get<string>(
+                            'GOOGLESHEET_RANGE_PHENOTYPES',
+                        ),
+                        configService.get<string>(
+                            'GOOGLESHEET_RANGE_IMPLICATIONS',
+                        ),
+                        configService.get<string>(
+                            'GOOGLESHEET_RANGE_RECOMMENDATIONS',
+                        ),
+                    ],
+                ),
+            ).rejects.toThrowError('Requested entity was not found.');
+        });
+
+        it('should return an error when sheet does not match structure', async () => {
+            expect(() =>
+                fetchSpreadsheetCells(
+                    configService.get<string>('EMPTY_GOOGLESHEET_ID'),
+                    configService.get<string>('GOOGLESHEET_APIKEY'),
+                    [
+                        configService.get<string>(
+                            'GOOGLESHEET_RANGE_MEDICATIONS',
+                        ),
+                        configService.get<string>('GOOGLESHEET_RANGE_GENES'),
+                        configService.get<string>(
+                            'GOOGLESHEET_RANGE_PHENOTYPES',
+                        ),
+                        configService.get<string>(
+                            'GOOGLESHEET_RANGE_IMPLICATIONS',
+                        ),
+                        configService.get<string>(
+                            'GOOGLESHEET_RANGE_RECOMMENDATIONS',
+                        ),
+                    ],
+                ),
+            ).rejects.toThrowError(/Unable to parse range.*/);
+        });
```

Comment:
These tests mostly test the Google Sheets API which is not necessary in my opinion. Removing them would also elevate the need for having a second empty testing sheet that's only used to check wether Google Sheets API returns the correct error message (which is also subject to change independently from our code).

Reply:
I agree, however, the tests also check, whether the `fetchSpreadsheetCells `-function propagates those errors instead of catching it inside the function. So I see some sense in the tests, but we could indeed consider removing them.

Reply:
My intention is writing them was similar to what Matti said. It seemed important to me to test how it is handeled when there is a wrong sheet ID or an unfitting structure. But if you say that this does not really matter because we trust the party deploying the server that they set the right ``.env`` variables, then I can delete them :)

Reply:
Not sure about the "trust the maintainer" part but even if we say we don't trust them and keep this test then we still have to trust them to run tests after changing their (gitignored) `.env`. I don't really mind, you can decide to keep or delete them :)

Reply:
Ok :3